plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -264,6 +264,115 @@ function ( string $content ): string {
+		$this->assertNotWPError( $attachment_id );
+
+		$image = wp_get_attachment_image(
+			$attachment_id,
+			'large',
+			false,
+			array(
+				'class' => 'wp-image-' . $attachment_id,
+				'alt'   => 'Green Leaves',
+			)
+		);
+
+		// Apply picture element support.
+		$this->opt_in_to_picture_element();
+
+		$picture_markup    = apply_filters( 'the_content', $image );
+		$picture_processor = new WP_HTML_Tag_Processor( $picture_markup );
+		$picture_processor->next_tag( array( 'tag_name' => 'SOURCE' ) );
+		$source_srcset = $picture_processor->get_attribute( 'srcset' );
+		$picture_processor->next_tag( array( 'tag_name' => 'IMG' ) );
+		$img_srcset = $picture_processor->get_attribute( 'srcset' );
+
+		// Extract file names from srcset.
+		$extract_file_names_from_srcset = static function ( string $srcset ): array {
+			$srcset_parts = explode( ', ', $srcset );
+			$files        = array();
+			foreach ( $srcset_parts as $srcset_part ) {
+				$parts   = explode( ' ', trim( $srcset_part ) );
+				$url     = $parts[0];
+				$files[] = pathinfo( basename( $url ), PATHINFO_FILENAME );
+			}
+			return $files;
+		};
+
+		$source_files = $extract_file_names_from_srcset( $source_srcset );
+		$img_files    = $extract_file_names_from_srcset( $img_srcset );
+
+		$this->assertCount( count( $source_files ), $img_files );
+
+		foreach ( $img_files as $img_index => $img_file ) {
+			// Check the file name starts with the same prefix.
+			// e.g. $img_file = 'leaves-350x200' and $source_files[ $img_index ] = 'leaves-350x350-jpg'.
+			$this->assertStringStartsWith( $img_file, $source_files[ $img_index ] );
+		}
+
+		wp_delete_attachment( $attachment_id, true );
+
+		if ( $tear_down instanceof Closure ) {
+			$tear_down();
+		}
+	}
+
+	/**
+	 * Data provider for test_picture_element_source_tag_srcset_has_same_image_sizes_as_img_tag_srcset.
+	 *
+	 * @return array<string, array{ set_up: Closure|null, tear_down: Closure|null }>
+	 */
+	public function data_provider_test_picture_element_source_tag_srcset_has_same_image_sizes_as_img_tag_srcset(): array {
+		return array(
+			'default_sizes' => array(
+				'set_up'    => null,
+				'tear_down' => null,
+			),
+			'when_two_different_image_sizes_have_same_width' => array(
+				'set_up'    => static function (): void {
+					add_image_size( 'square', 768, 768, true );
+					remove_all_filters( 'pre_option_medium_large_size_w' );
+					add_filter(
+						'pre_option_medium_large_size_w',
+						static function () {
+							return '768';
+						}
+					);
+					remove_all_filters( 'pre_option_medium_large_size_h' );
+					add_filter(
+						'pre_option_medium_large_size_h',
+						static function () {
+							return '512';
+						}
+					);
+				},
+				'tear_down' => static function (): void {
+					remove_image_size( 'square' );
+					remove_all_filters( 'pre_option_medium_large_size_w' );
+					remove_all_filters( 'pre_option_medium_large_size_h' );
+				},
```

Comment:
Since this is the only usage of the `tear_down` closure, we can probably work without them:
* Filters added in a test are removed automatically after the test, so I don't think we need to clean them up here.
* For the image size, maybe you can handle it in a `tear_down()` method on the class? Alternatively if it makes sense to keep it, we need to move when this `tear_down` closure is invoked in the test (see above comment).